Storage/SQL/BrowserDB.swift

```diff
@@ -203,65 +203,60 @@ open class BrowserDB {
             SentryIntegration.shared.sendWithStacktrace(message: "Unable to get a transaction: \(error.localizedDescription)", tag: "BrowserDB", severity: .error)
 
             // Check if the error we got is recoverable (e.g. SQLITE_BUSY, SQLITE_LOCK, SQLITE_FULL).
-            // If so, we *shouldn't* move the database file to a backup location and re-create it.
-            // Instead, just crash so that we don't lose any data.
+            // If so, we will try moving the database file to a backup location and copy it back to
+            // its original location. This should effectively "unlock" the database file.
```

Comment:
I don't get it.

You got `SQLITE_FULL`, indicating a full disk, so you try a backup (which fails), then move the file and move it back.

You got `SQLITE_LOCKED`, which indicates a coding error — _e.g._, simultaneously reading from the DB while dropping a table on the same connection — and we'll try the same. Probably the backup will succeed, the file will move, and move back, and it shouldn't affect the outcome. Though it might genuinely corrupt the database.

You got `SQLITE_BUSY`, which means another client is writing to or recovering the database, and you attempt to move the files out from under it. If the filesystem implements moves as renames, this'll do nothing at all — the file handles will be the same. If it does anything else, it might unpredictably corrupt the files. SQLite probably won't enjoy you moving the file while it's open — see errors like `SQLITE_READONLY_DBMOVED`.

I don't see attempting a move, then attempting a copy and replace, as being the correct course of action for any of these things. For `FULL`, you just need to abort (or try to clean up disk cache and try again). For `LOCKED`, we should close the connection and crash when we've done so, with a really descriptive Sentry message. For `BUSY`, the best thing we can do is wait — `BUSY` literally means that someone else is futzing with this very file, so it's not safe to copy!

And below this block of code, where we handle other kinds of errors, we should also be more circumspect. Some of those errors will indicate bugs in our code.